packages/firestore/src/model/values.ts

```diff
@@ -201,7 +210,24 @@ export function compare(left: api.Value, right: api.Value): number {
     case TypeOrder.NumberValue:
       return compareNumbers(left, right);
     case TypeOrder.TimestampValue:
-      return compareTimestamps(left.timestampValue!, right.timestampValue!);
+      if (isServerTimestamp(left)) {
+        if (isServerTimestamp(right)) {
+          return compareTimestamps(
+            getLocalWriteTime(left),
+            getLocalWriteTime(right)
+          );
+        } else {
+          // Server timestamps come after all concrete timestamps.
+          return 1;
```

Comment:
You know, if we just made `ServerTimestampValue` its own entry in the `TypeOrder` enum we wouldn't have to special case this at all: any Timestamp to ServerTimestamp comparison would be handled automatically by the way we handle differing types.

Reply:
That's @mikelehen level genius. I love it.

Reply:
WHAT DID I DO!?!?!?

Hi guys!